### PR TITLE
Scraping Song

### LINKED ISSUE #1132: Scraping Spotify
# Aim

Scraping and sending mails

# Details

Creating a list of popular songs of a particular artist from spotify and then mailing some interested users the list(links).

### PR DESCRIPTION
**Pull request by anumshka**
# Description

Scraping Songs from and mailing the list.

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] I have added the project meta data in the PR template.
- [x] I have created the ``requirements.txt`` file if needed.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [x] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [ ] Games
- [ ] Networking
- [ ] OS_Utilities
- [ ] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [ ] Fun
- [ ] Others

Title: ScrapingSongs

Folder: Scrapingsongs

Requirements: importing bs4,requests,smtplib,tabulate,MIMEText,MIMEMultipart

Script: ScrapingSongs.py

Arguments: None

Contributor: anumshka

Description: Scraping Songs from and mailing the list.

### CODE REVIEW

File: ScrapingSongs/ScrapingSongs.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import requests
+import pandas
+import smtplib
+import csv
+from tabulate import tabulate
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+# to get the html code
+response = requests.get("https://open.spotify.com/playlist/37i9dQZF1DX5KpP2LN299J")
```
Comment (kaustubhgupta) [inline ScrapingSongs/ScrapingSongs.py:11]: get the URL from user

File: ScrapingSongs/ScrapingSongs.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+song_names_spans = soup.find_all(name="span", class_="track-name")
+song_names = [song.getText() for song in song_names_spans]
+hyperlinks = soup.find_all(name="meta", property="music:song")
+song_links = [song.get("content") for song in hyperlinks]
+
+my_dict = {"Song Name": song_names,
+           "Song Link": song_links, }
+
+data = pandas.DataFrame(my_dict)
+data.to_csv("new_data.csv", index=False)
```
Comment (kaustubhgupta) [inline ScrapingSongs/ScrapingSongs.py:23]:
```suggestion
data.to_csv("./ScrapingSongs/new_data.csv", index=False)
```

File: ScrapingSongs/ScrapingSongs.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+html = """
+<html><body><p>Hello, Friend.</p>
+<p>Hope you enjoy this playlist:</p>
+{table}
+<p>Regards</p>
+</body></html>
+"""
+
+with open('new_data.csv', encoding='UTF-8') as input_file:
```
Comment (kaustubhgupta) [inline ScrapingSongs/ScrapingSongs.py:43]:
```suggestion
with open('./ScrapingSongs/new_data.csv', encoding='UTF-8') as input_file:
```

File: ScrapingSongs/ScrapingSongs.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+html = html.format(table=tabulate(data, headers="firstrow", tablefmt="html"))
+
+message = MIMEMultipart(
+    "alternative", None, [MIMEText(text), MIMEText(html, 'html')])
+
+# to send emails
+with smtplib.SMTP("smtp.gmail.com", port=587) as connection:
+    connection.starttls()
+    connection.login(user="sender_address", password="sender_password")
+    connection.sendmail(from_addr="sender_address", to_addrs="receiver_address", msg=message.as_string())
```
Comment (kaustubhgupta) [inline ScrapingSongs/ScrapingSongs.py:57]: Do you think this will send any email?

File: ScrapingSongs/new_data.csv [hunk 5]
```diff
@@ -0,0 +1,31 @@
+Song Name,Song Link
+Mr. Perfectly Fine (Taylor’s Version) (From The Vault),https://open.spotify.com/track/7l2tmgUhV7Y2aJHjiszifg
+Love Story (Taylor’s Version),https://open.spotify.com/track/3CeCwYWvdfXbZLXFhBrbnf
+willow,https://open.spotify.com/track/0lx2cLdOt3piJbcaXIV74f
+Blank Space,https://open.spotify.com/track/1u8c2t2Cy7UBoG4ArRcF5g
+champagne problems,https://open.spotify.com/track/0sY6ZUTh4yoctD8VIXz339
+cardigan,https://open.spotify.com/track/4R2kfaDFhslZEMJqAFNpdd
+Shake It Off,https://open.spotify.com/track/0cqRj7pUJDkTCEsJkx8snD
+Wildest Dreams,https://open.spotify.com/track/3fVnlF4pGqWI9flVENcT28
```
Comment (kaustubhgupta) [inline ScrapingSongs/new_data.csv:9]: delete this file

File: ScrapingSongs/readme.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Package/Script Name
```
Comment (kaustubhgupta) [inline ScrapingSongs/readme.md:1]: Is this the right heading for the readme file?

File: ScrapingSongs/readme.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+It scrapes spotify data to collect famous Taylor Swift's songs and mails it to her fans.
```
Comment (kaustubhgupta) [inline ScrapingSongs/readme.md:3]: Only Taylor swift and no other artists?

File: ScrapingSongs/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+## Setup instructions
+
+python file.py
```
Comment (kaustubhgupta) [inline ScrapingSongs/readme.md:7]: do you have any file named `file.py` ?

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by kaustubhgupta** [comment]
@anumshka update the PR before 2 pm so that we can review the code and merge it on time

**Comment by avinashkranjan** [comment]
Hey @anumshka are you going to update the PR..or Should I close it?

**Pull request closed without merging**